### PR TITLE
let configure check for supported c++11 flags and set -DVCL_CAN_STATIC_CONST_INIT_FLOAT accordingly

### PR DESCRIPTION
**Pull request by zarquon42b**
created autoconf based configure script.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -58,6 +58,8 @@ install:
   - export CXX=clang++
   - echo "CC=clang" >  ~/.R/Makevars
   - echo "CXX=clang++" >> ~/.R/Makevars
+  - echo "CXX1X=clang++" >> ~/.R/Makevars
+  - echo "CXX1XSTD=-std=c++11" >> ~/.R/Makevars
   - source ~/.R/Makevars
```
Comment (stnava) [inline .travis.yml]: thanks for this -

**Comment by zarquon42b** [comment]
What do think about using the configure script? This also makes the Makevars generation more CRAN compliant. And the test whether to use DVCL_CAN_STATIC_CONST_INIT_FLOAT =0 or 1 is done by trying to compile some example code.

**Comment by stnava** [comment]
first - i am thrilled that you have this working and i am making another attempt to address the unexplained OS X failure on travis though i am not hopeful.  i've looked this over a couple of times.  i don't know enough (near nil) about autoconf to understand what is happening here.  so i have a few questions:

(1)  are there other benefits to having this autoconf functionality available?

(2)  how challenging is this to maintain?  let us suppose i want to change the ants tag ( e.g. https://github.com/stnava/ANTsR/commit/bad1901d1ea0436e27e3510884136a983c55548b ), is this still a change that must be made in only one location?  or does it become two?

(3) what is the likelihood that these issues will continue in the future?

(4) i am not sure this is possible but is it better (i.e. easier to maintain, fewer lines of code ) to just generate a brief Makevars given detected  linux version?  

(5)  what is the new src/m4/vcl_cxx_static_const_init_float.m4?  why is this necessary?

anyway, forgive me if these are ignorant questions.

**Comment by zarquon42b** [comment]
(1) Maybe if there are more compatibility challenges, tests will be easier to implement
(2) simply replace the tag in configure.ac (https://github.com/zarquon42b/ANTsR/blob/autoconf-test/configure.ac#L39) and run ```autoreconf``` (on OSX you can run
brew install autoconf to install it)
(3) No idea, the issue is dependant on users and their config. At the moment the build, however, is broken for current Linux distributions.
(4) the issue is not linux or osx but the compiler version and its configuration: E.g. on ubuntu 14.04 with the clang3.4 it works with DVCL_CAN_STATIC_CONST_INIT_FLOAT =1 but fails on ubuntu 12.04 with the same clang version. I could not figure out a simple way to determine which flag to use other than to compile some example code.
(5) this is simply to keep configure.ac cleaner (using autoconf you usually create m4 files containing your  custom tests)

Nevermind, I only know about this stuff because CRAN forced me to do this in order to get my package Rvcg submitted.

**Comment by zarquon42b** [comment]
One thing: you never edit configure, only configure.ac (you can add posix compliant shell code as you like and on top can use a lot of handy macros  - e.g. check for header availability, etc)

**Comment by stnava** [comment]
ok - my fear is that this change will make antsr nearly impossible to maintain/compile for anyone other than dedicated people (me, you).   it also means additional system changing dependencies ( home-brew ).  perhaps i am in the minority, but i try to avoid home-brew as much as possible ... 

i am interested in learning these tricks but am not sure of the impact it will have on general usability.  i suppose the best deal would be to just auto-generate binaries on both OS X and linux via travis.  

anyway, i need to think more about this ... @jeffduda  - do you have any opinions on this?

almost seems like some of this is consequence of a clang bug/inconsistency.  

so it would seem unfortunate to add this complexity to antsr due to something that is likely to go away in the future.

**Comment by zarquon42b** [comment]
I understand your fear, however autoconf is a very commonly used and not that excentric ;). About homebrew: As I am not a OS X user (only running a VM for testing purposes), this was the fastest way to set up autoconf to fix the OS X issues (actually finding the typo).
The issue we are dealing with is **not** a bug, it is owed to simply different compiler configurations/versions regarding this feature (therefore the test whether it is supported properly, which would be the "clean" way to do this). However, if you do not want to use autoconf, you should cheat travis like that in .travis.yml (at least as long as travis uses 12.04):

```
- if [ "$TRAVIS_OS_NAME" == "linux" ]; then sed -i '-DVCL_CAN_STATIC_CONST_INIT_FLOAT=1|-DVCL_CAN_STATIC_CONST_INIT_FLOAT=0|' ANTsR/src/Makevars
fi
```

instead of breaking the install for the Linux users using current distributions (probably most of them). I will test other distributions like fedora next week. That way, the dirty hack is only in the travis config instead of in the packages Makevars file.

**Comment by zarquon42b** [comment]
I implemented the travis-cheat here: https://github.com/zarquon42b/ANTsR/tree/cheat-travis-vcl-can-static  and it builds correctly: https://travis-ci.org/zarquon42b/ANTsR/jobs/102560521.
However, I would suggest to add a note to README that compilers that do not support DVCL_CAN_STATIC_CONST_INIT_FLOAT=1 are not working and document the manual workaround (editing src/Makevars)

**Comment by zarquon42b** [comment]
>i suppose the best deal would be to just auto-generate binaries on both OS X and linux via travis. 

This is not a good idea for linux because there is no such thing as ***the*** linux OS (but many), as linux is only the kernel and the distributions ship with a large variety of system specific libraries/compilers etc.. E.g. ANTsR on recent Ubuntu versions do not compile with your ITKR binary. This is one cause why there are no linux binaries on CRAN.

**Comment by jeffduda** [comment]
I also try to avoid homebrew whenever possible and I definitely agree that
this further complicates maintenance/development, but it seems like this
(autoconf) might be a necessary evil in this case. I think at this point,
we definitely need to put together a developers manual that describes these
various issue and why thing are done the way they are. It would also help
for less complicated things like naming conventions, etc. It's much easier
to read one doc, than search though a bunch of old emails trying to figure
out what's going on.

**Pull request closed without merging**